Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,16 @@ def installed?
       !versions.empty?
     end
 
+    def latest_installed_version
+      versions.last
+    end
+
+    def outdated?(greedy)
+      (greedy || !auto_updates && !version == :latest) &&
+      !version.nil? &&  # ignore Casks without sources
```

Comment:
This is mostly just to make it more readable.

`(greedy || !auto_updates && !version == :latest)` is also pretty hard to follow. I'd probably write something like this: 

```ruby
unless greedy
  return false if auto_updates
  return false if version.latest?
end
```